akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -145,12 +150,15 @@ private[stream] object GraphInterpreter {
       i = 0
       while (i < connectionCount) {
         if (ins(i) ne null) {
-          inHandlers(i) = logics(inOwners(i)).inHandlers(ins(i).id)
-          logics(inOwners(i)).inToConn(ins(i).id) = i
+          val logic = logics(inOwners(i))
+          inHandlers(i) = logic.handlers(ins(i).id).asInstanceOf[InHandler]
+          logics(inOwners(i)).portToConn(ins(i).id) = i
         }
         if (outs(i) ne null) {
-          outHandlers(i) = logics(outOwners(i)).outHandlers(outs(i).id)
-          logics(outOwners(i)).outToConn(outs(i).id) = i
+          val logic = logics(outOwners(i))
+          val inCount = logic.inCount
+          outHandlers(i) = logic.handlers(outs(i).id + inCount).asInstanceOf[OutHandler]
```

Comment:
there are no longer separate `inHandlers` and `outHandlers`, they are all stored in the same array to reduce object header overhead.